src/FluentValidation/DefaultValidatorOptions.cs

```diff
@@ -382,6 +382,38 @@ public static class DefaultValidatorOptions {
 			return rule.Configure(config => config.CurrentValidator.Options.Severity = severity);
 		}
 
+		/// <summary>
+		/// Specifies custom severity that should be stored alongside the validation message when validation fails for this rule.
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		/// <typeparam name="TProperty"></typeparam>
+		/// <param name="rule"></param>
+		/// <param name="severityProvider"></param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, TProperty> WithSeverity<T, TProperty>(this IRuleBuilderOptions<T, TProperty> rule, Func<T, Severity> severityProvider) {
+			severityProvider.Guard("A lambda expression must be passed to WithSeverity", nameof(severityProvider));
+			var wrapper = new Func<PropertyValidatorContext, Severity>(ctx => severityProvider((T) ctx.Instance));
+			return rule.Configure(config => config.CurrentValidator.Options.CustomSeverityProvider = wrapper);
+		}
+
+		/// <summary>
+		/// Specifies custom severity that should be stored alongside the validation message when validation fails for this rule.
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		/// <typeparam name="TProperty"></typeparam>
+		/// <param name="rule"></param>
+		/// <param name="severityProvider"></param>
+		/// <returns></returns>
+		public static IRuleBuilderOptions<T, TProperty> WithSeverity<T, TProperty>(this IRuleBuilderOptions<T, TProperty> rule, Func<T, TProperty, Severity> severityProvider) {
+			severityProvider.Guard("A lambda expression must be passed to WithSeverity", nameof(severityProvider));
+
+			var wrapper = new Func<PropertyValidatorContext, Severity>(ctx => {

```

Comment:
You can use a c#7 local function to make this nicer & easier to read.

Reply:
I just copied the code from `WithState`.
That should be changed then too i guess.

Reply:
To be honest, i don't fully understand what should be changed to a local function here. Could you give me a sample for this case?

Reply:
I meant as an alternative to the `wrapper` variable. Not really necessary but the syntax is slightly nicer than instantiating a `Func`

```csharp
Severity SeverityProvider(PropertyValidatorContext ctx) {
	return severityProvider((T) ctx.Instance, (TProperty) ctx.PropertyValue);
}

return rule.Configure(config => config.CurrentValidator.Options.CustomSeverityProvider = SeverityProvider);
```

Reply:
Done